networking_cisco/tests/unit/cisco/cfg_agent/test_aci_asr1k_cfg_syncer.py

```diff
@@ -0,0 +1,112 @@
+          ' nrouter-3ea5f9_nat_pool vrf nrouter-3ea5f9 overload',
+         u'ip nat pool nrouter-3ea5f9_nat_pool 172.16.0.124'
+          ' 172.16.0.124 netmask 255.255.0.0',
+         u'ip route vrf nrouter-3ea5f9 0.0.0.0 0.0.0.0'
+          ' Port-channel10.3000 172.16.0.1',
+         u'ip access-list standard neutron_acl_2564',
+         <IOSCfgLine # 83 'interface Port-channel10.2564'>,
+         <IOSCfgLine # 96 'interface Port-channel10.3000'>,
+         u'nrouter-3ea5f9']
+        """
+        cfg.CONF.set_override('enable_multi_region', False, 'multi_region')
+
+        router_db_info = []
+
+        self.config_syncer = aci_asr1k_cfg_syncer.ConfigSyncer(router_db_info,
+                                                      self.driver,
+                                                      self.hosting_device_info)
+        self.config_syncer.get_running_config = mock.Mock(
+            return_value=self._read_asr_running_cfg(
+                               'asr_basic_running_cfg_no_multi_region.json'))
+
+        invalid_cfg = self.config_syncer.delete_invalid_cfg()
+        self.assertEqual(8, len(invalid_cfg))
+
+    def test_delete_invalid_cfg_with_multi_region_and_empty_routers_list(self):
+        """
+        This test verifies that the  cfg-syncer will delete invalid cfg
+        if the neutron-db (routers dictionary list) happens to be empty.
+
+        Since the neutron-db router_db_info is empty, all region 0000002
+        running-config should be deleted.
+
+        Expect 8 invalid configs found
+
+        ['ip nat inside source static 10.2.0.5 172.16.0.126'
+          ' vrf nrouter-3ea5f9-0000002 redundancy neutron-hsrp-1064-3000',
+         'ip nat inside source list neutron_acl_0000002_2564 pool '
+         'nrouter-3ea5f9-0000002_nat_pool vrf nrouter-3ea5f9-0000002 overload',
+         'ip nat pool nrouter-3ea5f9-0000002_nat_pool '
+          '172.16.0.124 172.16.0.124 netmask 255.255.0.0',
+         'ip route vrf nrouter-3ea5f9-0000002 0.0.0.0 0.0.0.0'
+          ' Port-channel10.3000 172.16.0.1',
+         'ip access-list standard neutron_acl_0000002_2564',
+         <IOSCfgLine # 83 'interface Port-channel10.2564'>,
+         <IOSCfgLine # 96 'interface Port-channel10.3000'>,
+         'nrouter-3ea5f9-0000002']
+        """
+        cfg.CONF.set_override('enable_multi_region', True, 'multi_region')
+        cfg.CONF.set_override('region_id', '0000002', 'multi_region')
+        cfg.CONF.set_override('other_region_ids', ['0000001'], 'multi_region')
+        router_db_info = []
+        self.config_syncer = aci_asr1k_cfg_syncer.ConfigSyncer(router_db_info,
+                                                      self.driver,
+                                                      self.hosting_device_info)
+        self.config_syncer.get_running_config = \
+            mock.Mock(return_value=self._read_asr_running_cfg(
+                                            'asr_basic_running_cfg.json'))
```

Comment:
this asr_basic_running_cfg.json file doesn't contain config items tagged with region 0000001 so this UT doesn't really test the functionality that it will keep other region's config items....